functions/src/helpers/handle-referral.ts

```diff
@@ -50,22 +50,15 @@ export async function handleReferral(user: User, eventId: string) {
     }
     console.log(`referredByGroup: ${referredByGroup}`)
 
-    const txns = (
-      await firestore
+    const txns = await transaction.get(
+      firestore
         .collection('txns')
         .where('toId', '==', referredByUserId)
         .where('category', '==', 'REFERRAL')
-        .get()
-    ).docs.map((txn) => txn.ref)
-    if (txns.length > 0) {
-      const referralTxns = await transaction.getAll(...txns).catch((err) => {
-        console.error('error getting txns:', err)
-        throw err
-      })
+    )
+    if (txns.size > 0) {
       // If the referring user already has a referral txn due to referring this user, halt
-      if (
-        referralTxns.map((txn) => txn.data()?.description).includes(user.id)
-      ) {
+      if (txns.docs.map((txn) => txn.data()?.description).includes(user.id)) {
```

Comment:
This is still a bad piece of code; we should add the referral user ID directly to the referral txn object so that we can filter on it in the query.

Reply:
Yep